experimental/dds/tree2/src/test/shared-tree-core/branch.spec.ts

```diff
@@ -439,6 +435,54 @@ describe("Branches", () => {
 		assert.equal(branch.getHead().revision, nullRevisionTag);
 	});
 
+	describe("all nested forks and transactions are disposed and aborted when transaction is", () => {
+		const setUpNestedForks = (rootBranch: DefaultBranch) => {
+			change(rootBranch);
+			const fork1 = rootBranch.fork();
+			change(rootBranch);
+			rootBranch.startTransaction();
+			const fork2 = rootBranch.fork();
+			change(rootBranch);
+			const fork3 = rootBranch.fork();
+			change(fork3);
+			fork3.startTransaction();
+			change(fork3);
+			const fork4 = fork3.fork();
+
+			return {
+				disposedFroks: [fork2, fork3, fork4],
```

Comment:
```suggestion
				disposedFroks: [fork2, fork3, fork4, fork4.fork()],
```

This case is not handled. Namely, if a branch is in a transaction, and it forks, and _that_ fork forks, we still want to dispose both forks even if the first fork is not transacting. This is because the second fork will be referring to a commit that no longer exists on the original branch after the squash.

You could try using `onForkTransitive` to track all transitive branches.

Reply:
This was the point of [this comment](https://github.com/microsoft/FluidFramework/pull/16493/files#r1270828378). I might be missing something but it looks like it was resolved without being addressed.

Reply:
I didn't see any of your comments until just now, you did indeed say the exact same thing. However, I still don't think it is working properly.